pkg/policyreport/changerequestcreator.go

```diff
@@ -21,103 +18,60 @@ import (
 // merges and creates requests every tickerInterval
 type creator interface {
 	add(request *unstructured.Unstructured)
-	create(request *unstructured.Unstructured) error
 	run(stopChan <-chan struct{})
 }
 
 type changeRequestCreator struct {
-	client versioned.Interface
-
-	// addCache preserves requests that are to be added to report
-	RCRCache *cache.Cache
-
-	CRCRCache *cache.Cache
-	// removeCache preserves requests that are to be removed from report
-	// removeCache *cache.Cache
-	mutex sync.RWMutex
-	queue []string
-
+	client         versioned.Interface
+	mutex          *sync.RWMutex
+	queue          []*unstructured.Unstructured
 	tickerInterval time.Duration
-
-	log logr.Logger
+	log            logr.Logger
 }
 
 func newChangeRequestCreator(client versioned.Interface, tickerInterval time.Duration, log logr.Logger) creator {
 	return &changeRequestCreator{
 		client:         client,
-		RCRCache:       cache.New(0, 24*time.Hour),
-		CRCRCache:      cache.New(0, 24*time.Hour),
-		queue:          []string{},
+		mutex:          &sync.RWMutex{},
+		queue:          []*unstructured.Unstructured{},
```

Comment:
I thought we were using the rate limit queue for retries, but it doesn't seem to be the case. There should be no failure in merging change requests correct?

Reply:
> There should be no failure in merging change requests correct?

Not sure what you mean here. I don't think I changed the error logic, only the storage.
Can you elaborate ?

Reply:
We used to use the rate limit queue in a controller to retry on failures. But I don't see any failure returned in this controller, is that the case? If so, I'm good with the change.